lib/pbench/server/api/resources/upload_api.py

```diff
@@ -128,96 +149,124 @@ def put(self, controller):
+            self.logger.info(
+                "Dataset already exists, user = {}, ctrl = {!a}, file = {!a}",
+                username,
+                controller,
+                filename,
+            )
             response = jsonify(dict(message="Dataset already exists"))
-            response.status_code = 200
+            response.status_code = HTTPStatus.OK
             return response
         except Exception:
-            self.logger.exception("unable to create dataset for {}", filename)
+            self.logger.exception(
+                "unable to create dataset for user = {}, ctrl = {!a}, file = {!a}",
+                username,
+                controller,
+                filename,
+            )
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
         if tar_full_path.is_file() or md5_full_path.is_file():
             self.logger.error(
-                "Dataset or corresponding md5 file already present on the disc, {}",
-                filename,
+                "Dataset, or corresponding md5 file, already present; tar {} ({}), md5 {} ({})",
+                tar_full_path,
+                "present" if tar_full_path.is_file() else "missing",
+                md5_full_path,
+                "present" if md5_full_path.is_file() else "missing",
             )
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
-        self.logger.info("Uploading file {} to {}", filename, dataset)
+        self.logger.info(
+            "Uploading file {!a} (user = {}, ctrl = {!a}) to {}",
+            filename,
+            username,
+            controller,
+            dataset,
+        )
 
         with tempfile.NamedTemporaryFile(mode="wb", dir=path) as ofp:
-            chunk_size = 4096
-            self.logger.debug("Writing chunks")
             hash_md5 = hashlib.md5()
 
             try:
                 while True:
-                    chunk = request.stream.read(chunk_size)
+                    chunk = request.stream.read(self.CHUNK_SIZE)
                     bytes_received += len(chunk)
                     if len(chunk) == 0 or bytes_received > content_length:
                         break
 
                     ofp.write(chunk)
                     hash_md5.update(chunk)
             except Exception:
+                msg = "Unexpected error encountered during file upload"
                 self.logger.exception(
-                    "Tarfile upload: There was something wrong uploading {}", filename
+                    "{} for user = {}, ctrl = {!a}, file = {!a}",
+                    username,
+                    controller,
+                    filename,
                 )
-                abort(500, message=f"There was something wrong uploading {filename}")
+                abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
```

Comment:
LGTM is complaining about multiple assignments to `msg` here. I'm assuming it's because it doesn't understand `abort`; but it's true that with the code as it is, this assignment to `msg` isn't used at all before the `abort`. Turns out that it's wrong but it's right anyway: you're missing the first parameter in the `self.logger.exception` here!